coherence-spring-core/src/test/java/com/oracle/coherence/spring/event/MapEventHandlerTests.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2013, 2025, Oracle and/or its affiliates.
+ *
+ * Licensed under the Universal Permissive License v 1.0 as shown at
+ * https://oss.oracle.com/licenses/upl.
+ */
+package com.oracle.coherence.spring.event;
+
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ForkJoinPool;
+
+import com.oracle.bedrock.testsupport.deferred.Eventually;
+import com.oracle.coherence.spring.CoherenceServer;
+import com.oracle.coherence.spring.annotation.Name;
+import com.oracle.coherence.spring.annotation.SessionName;
+import com.oracle.coherence.spring.annotation.event.MapName;
+import com.oracle.coherence.spring.annotation.event.Synchronous;
+import com.oracle.coherence.spring.configuration.annotation.CoherenceCache;
+import com.oracle.coherence.spring.configuration.annotation.EnableCoherence;
+import com.oracle.coherence.spring.configuration.session.SessionConfigurationBean;
+import com.oracle.coherence.spring.configuration.session.SessionType;
+import com.tangosol.net.NamedCache;
+import com.tangosol.net.Session;
+import com.tangosol.util.InvocableMap;
+import com.tangosol.util.MapEvent;
+import data.Person;
+import data.PhoneNumber;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+
+@SpringJUnitConfig(MapEventHandlerTests.Config.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class MapEventHandlerTests {
+
+	@Inject
+	ConfigurableApplicationContext context;
+
+	@Inject
+	@Name("test")
+	private Session session;
+
+	@Inject
+	private TestObservers observers;
+
+	@Inject
+	private CoherenceServer coherenceServer;
+
+	@CoherenceCache
+	@SessionName("test")
+	private NamedCache<String, Person> people;
+
+	@Test
+	@DirtiesContext
+	void testEventInterceptorMethods() {
+		this.observers.asyncMethodEvents.clear();
+		this.observers.syncMethodEvents.clear();
+		// saturate ForkJoinPool to make listener registration slow
+		for (int i = 0; i < 8; i++) {
```

Comment:
As you stated to me privately, 8 is the number of cores on your system.  But on a larger system, this likely won't be sufficient.  Maybe we should get the number of course, and then pad the number to queue up enough tasks to ensure we never get access to the FJP before we should